python/src/rules_engine/parser.py

```diff
@@ -93,19 +129,41 @@ def _newline_line_ending(data: str) -> str:
     return data
 
 
+def _remove_double_quotes(data: str) -> str:
+    """Return a string with the double quotes removed."""
+    return data.replace('"', "")
+
+
+def _replace_tabs_with_commas(data: str) -> str:
+    """Return a a string with tabs replaced by commas."""
+    return data.replace("\t", ",")
+
+
+def _remove_non_ascii_or_newline_characters(data: str) -> str:
+    """Remove all non-ASCII, non-newline characters"""
+    return "".join(
+        character for character in data if character in _ASCII_CHARACTERS_PLUS_NEWLINE
+    )
+
+
 def _detect_gas_company(data: str) -> NaturalGasCompany:
     """
     Return which natural gas company issued this bill.
     """
+    header = data.split("\n")[0]
+    date_exists = "End Date" in header or "Read Date" in header
+    days_exist = "Days In Bill" in header or "Days" in header
+    therms_exist = "Usage" in header or "Usage (Therms)" in header
+    is_eversource = date_exists and days_exist and therms_exist
+
     if _NaturalGasCompanyBillRegex.NATIONAL_GRID.search(data):
         return NaturalGasCompany.NATIONAL_GRID
-    elif _NaturalGasCompanyBillRegex.EVERSOURCE.search(data):
+    elif is_eversource:
         return NaturalGasCompany.EVERSOURCE
     else:
-        raise ValueError(
-            """Could not detect which company this bill was from:\n 
-                           Regular expressions matched not."""
-        )
+        print("bad header")
+        print(header)
+        raise ValueError("""Could not detect which company this bill was from""")
```

Comment:
TODO: Catch python errors in js and show a nice invalid input warning from conform